learnosity_sdk/request/init.py

```diff
@@ -38,20 +40,16 @@ def __init__(
         self.request = request.copy()
         self.action = action
 
-        self.request_string = self.generate_request_string()
+        self.sign_request_data = True
```

Comment:
I think we need to call these things in this order:
1. `validate()` - which does validation and converts the request to a dict if it's passed as a string
2. `add_telemetry_data()` - to update the request now that it's guaranteed to be a dict
3. `generate_request_string()` - create the string after adding the telemetry data
4. Set the initial value for `sign_request_data` because it's updated by:
5. `set_service_options()` in some cases.